From cca6a28269b9629bba9710823bcc4536c0e1b7af Mon Sep 17 00:00:00 2001
From: Philipp Bartsch <phil@grmr.de>
Date: Thu, 9 Oct 2025 20:59:41 +0200
Subject: [PATCH] Fix building unit gtest on darwin

Looks like a result of -Werror and an overly cautious compiler warning.

/private/tmp/nix-build-vectorscan-5.4.12.drv-0/source/unit/gtest/gtest-all.cc:7485:26: error: variable 'dummy' is uninitialized when passed as a const pointer argument here [-Werror,-Wuninitialized-const-pointer]
 7485 |   StackLowerThanAddress(&dummy, &result);
      |                          ^~~~~

Found by hydra building vectorscan-5.4.12 for aarch64-darwin and x86_64-darwin.

https://hydra.nixos.org/build/309203585
https://hydra.nixos.org/build/309203586
---
 unit/gtest/gtest-all.cc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/unit/gtest/gtest-all.cc b/unit/gtest/gtest-all.cc
index a9a03b2e3..25a6ff635 100644
--- a/unit/gtest/gtest-all.cc
+++ b/unit/gtest/gtest-all.cc
@@ -7480,7 +7480,7 @@ void StackLowerThanAddress(const void* ptr, bool* result) {
 }
 
 bool StackGrowsDown() {
-  int dummy;
+  int dummy = 0;
   bool result;
   StackLowerThanAddress(&dummy, &result);
   return result;
